### PR TITLE
Add media-feature-name-black(white)list rules

### LINKED ISSUE #1722: Add media-feature-name-*list rules
- [ ] `media-feature-name-blacklist`
- [ ] `media-feature-name-whitelist`

This proposal comes via a [Stack Overflow support request](http://stackoverflow.com/questions/38572426/how-to-use-stylelint-to-prevent-use-of-max-width-in-media-queries) that I found no other way to solve.

Two new rules `media-feature-blacklist` & `media-feature-whitelist`

Specify a blacklist/whitelist of disallowed/allowed media features:

```css
@media (max-width: 50em) { padding: 1rem; }
/**     ↑
 *       Media features like this */
```

Options

`array|string`: `["array", "of", "unprefixed", "media-features"]|"media-feature"`

Given:

```js
  "media-feature-blacklist": [
    "max-width",
  ],
  "media-feature-whitelist": [
    "min-width",
  ],
```

The following patterns are considered warnings:

```css
@media (max-width: 50em) {
  padding: 1rem;
}
```

The following patterns are *not* considered warnings:
```css
@media (min-width: 50em) {
  padding: 1rem;
}
```

----

Reference: MDN [Media Features](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Media_features)

### PR DESCRIPTION
**Pull request by gaidarenko**
Closes #1722

### CODE REVIEW

File: src/rules/media-feature-name-blacklist/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    message: messages.rejected("max-width"),
+    line: 1,
+    column: 52,
+  }, {
+    code: "@media (my-width: 50em) { padding: 1rem; }",
+    message: messages.rejected("my-width"),
+    line: 1,
+    column: 9,
+  } ],
+})
```
Comment (jeddy3) [inline src/rules/media-feature-name-blacklist/__tests__/index.js]: Can we make sure that media feature names containing SCSS & Less variables and operators are ignored?

File: src/rules/media-feature-name-blacklist/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    "width",
+    "/^my-/",
+  ]],
+
+  accept: [ {
+    code: "@media (min-width: 50em) { padding: 1rem; }",
+  }, {
+    code: "@media (min-width: 50em) { max-width: 50em; }",
+  }, {
+    code: "@media (--max-width: 50em) { padding: 1rem; }",
```
Comment (jeddy3) [inline src/rules/media-feature-name-blacklist/__tests__/index.js]: I'm not sure this is valid.

*Custom properties* are only found in declaration blocks. *Custom media queries* have [no value](https://www.w3.org/TR/2016/WD-mediaqueries-4-20160126/#custom-mq) e.g.

```css
@media (--wide-viewport) {}
```

I see you're already correctly ignoring custom media queries, and so I'd remove `--max-width` from the config

File: src/rules/media-feature-name-blacklist/__tests__/index.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    description: "ignore custom media query",
+  }, {
+    code: "@media (/* max-width: 50em */ min-width: 50em) { padding: 1rem; }",
+    description: "ignore comments",
+  }, {
+    code: "@media (width <= 50em) { padding: 1rem; }",
```
Comment (jeddy3) [inline src/rules/media-feature-name-blacklist/__tests__/index.js]: Can we also get a test for a nested range context e.g.

```css
@media (400px < width < 1000px) {}
```

File: src/rules/media-feature-name-blacklist/index.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+﻿import {
```
Comment (jeddy3) [inline src/rules/media-feature-name-blacklist/index.js]: Looks good! :)

File: src/rules/media-feature-name-whitelist/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+Given:
+
+```js
+["max-width", "/^my-/"]
+```
+
+The following patterns are considered warnings:
+
+```css
+@media (min-width: 50em) {
```
Comment (jeddy3) [inline src/rules/media-feature-name-whitelist/README.md]: Remove declarations

File: src/rules/media-feature-name-whitelist/__tests__/index.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  ]],
+
+  accept: [ {
+    code: "@media (max-width: 50em) { padding: 1rem; }",
+  }, {
+    code: "@media (MaX-wIdTh: 50em) { padding: 1rem; }",
+  }, {
+    code: "@media (max-width: 50em) { min-width: 50em; }",
+  }, {
+    code: "@media (--min-width: 50em) { padding: 1rem; }",
```
Comment (jeddy3) [inline src/rules/media-feature-name-whitelist/__tests__/index.js]:
```
@media (--wide-viewport) {}
```

File: src/rules/media-feature-name-whitelist/__tests__/index.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+testRule(rule, {
+  ruleName,
+  config: [[
+    "max-width",
+    "/^my-/",
+  ]],
+
+  accept: [ {
+    code: "@media (max-width: 50em) { padding: 1rem; }",
```
Comment (jeddy3) [inline src/rules/media-feature-name-whitelist/__tests__/index.js]: Remove all declarations.

File: src/rules/media-feature-name-blacklist/__tests__/index.js (hunk 2 above)
Comment (jeddy3) [inline src/rules/media-feature-name-blacklist/__tests__/index.js]: Ok.

**Comment by gaidarenko** [comment]
@jeddy3 Done.

**Comment by jeddy3** [comment]
@gaidarenko Excellent work!

Added to changelog as:

-   Added: `media-feature-name-blacklist` rule.
-   Added: `media-feature-name-whitelist` rule.

**Pull request merged**